gooddata-java/src/main/java/com/gooddata/sdk/service/gdc/DataStoreService.java

```diff
@@ -95,7 +99,10 @@ public void upload(String path, InputStream stream) {
 
     private void upload(URI url, InputStream stream) {
         try {
-            sardine.put(url.toString(), stream);
+            // We need to use it this way, if we want to track request_id in the stacktrace.
+            InputStreamEntity entity = new InputStreamEntity(stream, -1);
```

Comment:
is there some name for `-1` magic number?

Reply:
Named and added comment.

Reply:
well as far as I googled the javadoc: https://appdoc.app/artifact/org.apache.httpcomponents/httpcore/4.4.6/org/apache/http/entity/InputStreamEntity.html#InputStreamEntity(java.io.InputStream,%20long)

`-1` means unknown length and it's the default - so please rather just use the stream-only constructor

Reply:
Used stream-only constructor.